test/stage-options.test.ts

```diff
@@ -202,56 +217,281 @@ test('can set pre/post github action job step', () => {
+          if: stageBOptions.jobSettings?.if,
+        },
+      },
+    });
+  });
+});
+
+test('github stages in waves works', () => {
+  withTemporaryDirectory((dir) => {
+    const buildIfStatement =
+      "contains(github.event.issue.labels.*.name, 'deployToA') || contains(github.event.issue.labels.*.name, 'deployToB')";
+    const pipeline = new GitHubWorkflow(app, 'Pipeline', {
+      workflowPath: `${dir}/.github/workflows/deploy.yml`,
+      synth: new ShellStep('Build', {
+        installCommands: ['yarn'],
+        commands: ['yarn build'],
+      }),
+      jobSettings: {
+        if: buildIfStatement,
+      },
     });
 
+    const stageAOptions: GitHubStageProps = {
+      jobSettings: {
+        if: "success() && contains(github.event.issue.labels.*.name, 'deployToA')",
+      },
+    };
+    const stageA = new GitHubStage(app, 'MyStageA', {
+      env: { account: '111111111111', region: 'us-east-1' },
+      ...stageAOptions,
+    });
+
+    new Stack(stageA, 'MyStackA');
+
+    const stageBOptions: GitHubStageProps = {
+      env: { account: '12345678901', region: 'us-east-1' },
+      jobSettings: {
+        if: "success() && contains(github.event.issue.labels.*.name, 'deployToB')",
+      },
+    };
+    const stageB = new GitHubStage(app, 'MyStageB', stageBOptions);
+
+    new Stack(stageB, 'MyStackB');
+
+    // Make a wave to have the stages be parallel (not depend on each other)
+    const wave = pipeline.addWave('MyWave', {
+      pre: [
+        new GitHubActionStep('PreWaveAction', {
+          jobSteps: [
+            {
+              name: 'pre wave action',
+              uses: 'my-pre-wave-action@1.0.0',
+              with: {
+                'app-id': 1234,
+                'secrets': 'my-secrets',
+              },
+            },
+          ],
+        }),
+      ],
+
+      post: [
+        new GitHubActionStep('PostWaveAction', {
+          jobSteps: [
+            {
+              name: 'Checkout',
+              uses: 'actions/checkout@v3',
+            },
+            {
+              name: 'post wave action',
+              uses: 'my-post-wave-action@1.0.0',
+              with: {
+                'app-id': 4321,
+                'secrets': 'secrets',
+              },
+            },
+          ],
+        }),
+      ],
+    });
+    wave.addStage(stageA);
+    wave.addStage(stageB);
+
+    app.synth();
+
+    const workflowFileContents = readFileSync(pipeline.workflowPath, 'utf-8');
+    expect(workflowFileContents).toMatchSnapshot();
+
+    const yaml = YAML.parse(workflowFileContents);
+    expect(yaml).toMatchObject({
+      jobs: {
+        'Build-Build': {
+          if: buildIfStatement,
+        },
+        'MyWave-MyStageA-MyStackA-Deploy': {
+          if: stageAOptions.jobSettings?.if,
+        },
+        'MyWave-MyStageB-MyStackB-Deploy': {
+          if: stageBOptions.jobSettings?.if,
+        },
+      },
+    });
+  });
+});
+
+test('stages in pipeline works with `if`', () => {
```

Comment:
is this test still relevant in the PR? How is it succeeding?

Reply:
Yes. I modified it to track the current behavior. It appears that `if` added to different stages tracks those stages properly.